feat: V2 fields in chara editor (flagged), V2 import, V2 export (with V1 fields backfilled)

malfoyslastname opened the pull request:
addresses part of #330 

- When the V2 flag is active (type `flag('charv2')`), the character creator/editor will now show card V2 fields (NOTE: It is not possible yet to edit a Character Book that was bundled with a downloaded character card, in any way. You can only swap it out for an Agnai Memory Book.
- Regardless of flag, it is now possible to import V2 cards
- Regardless of flag, exporting in Tavern format will export as a V2 card with V1 fields backfilled. Agnai voice setting is saved under "agnai" extension.
- Tavern is now the default export format. The only data lost upon converting Agnai cards to Tavern is creation date, update date, and favorite status, so after field testing we might want to drop support for exporting in Agnai format. 
- Plain text is now the default Persona format
- Native memory book has been extended to include Character Book spec fields. We don't need to *use* them, we just need to *not destroy* them.

V2 fields are NOT yet used in actual chats, they're only importable/modifiable/exportable.

Some QA done by me but further QA desirable.

Commit:
feat: V2 fields in chara editor (flagged), V2 import, V2 export (with V1 fields backfilled)

- When the V2 flag is active (type `flag('charv2')`), the character creator/editor will now show card V2 fields (NOTE: It is not possible yet to edit a Character Book that was bundled with a downloaded character card, in any way. You can only swap it out for an Agnai Memory Book.
- Regardless of flag, it is now possible to import V2 cards
- Regardless of flag, exporting in Tavern format will export as a V2 card with V1 fields backfilled. Agnai voice setting is saved under "agnai" extension.
- Tavern is now the default export format. The only data lost upon converting Agnai cards to Tavern is creation date, update date, and favorite status, so after field testing we might want to drop support for exporting in Agnai format. (We could even export those missing fields inside the agnai extension if deemed sensible!)
- Plain text is now the default Persona format
- Native memory book has been extended to include Character Book spec fields. We don't need to *use* them, we just need to *not destroy* them.

V2 fields are NOT yet used in actual chats, they're only importable/modifiable/exportable.

s

sceuick commented on web/pages/Character/CreateCharacter.tsx:
```diff
@@ -84,6 +72,11 @@ const CreateCharacterV1: Component = () => {
     }
   })
+  const memory = memoryStore()
+  const initialBook = () => state.edit?.characterBook
```
Gotta use `createMemo` for anything like this.

sceuick committed:
Minor refactors
- Memoize in createcharacter
- use functors in memory functions
- move char functions to common char module

Commit:
add plaintext personality placeholder

Commit:
add missing createMemo and use better magic string for none value
